testing/dbtest/template.go

```diff
@@ -0,0 +1,111 @@
+package dbtest
+
+import (
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"time"
+
+	// postgres dialect
+	_ "github.com/lib/pq"
+)
+
+var (
+	// StartUsingTemplate creates a new test database from a postgres template database.
+	StartUsingTemplate func(dialect string, opt ...Option) (func() error, string, string, error) = startUsingTemplate
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyz"
+const (
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+func randStr(n int) string {
+	b := make([]byte, n)
+	// A rand.Int63() generates 63 random bits, enough for letterIdxMax letters!
+	for i, cache, remain := n-1, rand.Int63(), letterIdxMax; i >= 0; {
+		if remain == 0 {
+			cache, remain = rand.Int63(), letterIdxMax
+		}
+		if idx := int(cache & letterIdxMask); idx < len(letterBytes) {
+			b[i] = letterBytes[idx]
+			i--
+		}
+		cache >>= letterIdxBits
+		remain--
+	}
+
+	return string(b)
+}
+
+func startUsingTemplate(dialect string, opt ...Option) (cleanup func() error, retURL, dbname string, err error) {
+	db, err := sql.Open("postgres", "postgres://boundary:boundary@127.0.0.1/boundary?sslmode=disable")
```

Comment:
Is an empty dialect a valid parameter?

Reply:
It is not, and really it should only be one of the supported dialects (so it should only be `postgres` currently). I can add a check along with defining the constants you mention in another comment.

Reply:
Added a check for this